cpp/mrc/include/mrc/edge/edge_builder.hpp

```diff
@@ -176,6 +176,89 @@ struct EdgeBuilder final
         sink.set_readable_edge_handle(edge);
     }
 
+    template <typename EdgeDataTypeT, typename SourceT, typename SinkT, typename SpliceInputT, typename SpliceOutputT>
+    static void make_edge_splice(SourceT& source, SinkT& sink, SpliceInputT& splice_input, SpliceOutputT& splice_output)
+    {
+        using source_full_t = SourceT;
+        using sink_full_t   = SinkT;
+
+        // Have to jump through some hoops here, mimics what 'writable_acceptor_typed' does, so we get everything
+        // aligned correctly.
+        if constexpr (is_base_of_template<edge::IWritableAcceptor, source_full_t>::value &&
+                      is_base_of_template<edge::IWritableProvider, sink_full_t>::value)
+        {
+            /*
+             * In this case, the source object has accepted a writable edge from the sink.
+             *
+             * Given: [source [edge_handle]] -> [sink]
+             *
+             * We will:
+             * - Get a reference to the edge_handle the source is holding
+             * - Reset the Source edge connection
+             * - Create a new edge from the source to our WritableProvider splice node
+             * - Set the edge_handle for the WritableAcceptor splice node to the edge_handle from the source
+             *
+             * This will result in the following:
+             *
+             * [source[new_edge_handle]] -> [splice_node[old_edge_handle]] -> [sink]
+             *
+             */
+            // We don't need to know the data type of the sink, the source node will have the same data type as the
+            // splice node, and we already know the sink can provide an edge for the source's data type.
+            // [source] -> [sink] => [[source] -> [splice_node]] -> [sink]
+            auto* splice_writable_provider = dynamic_cast<edge::IWritableProvider<EdgeDataTypeT>*>(&splice_input);
+            CHECK(splice_writable_provider != nullptr) << "Splice input is not a writable provider";
+
+            auto* splice_writable_acceptor = dynamic_cast<edge::IWritableAcceptor<EdgeDataTypeT>*>(&splice_output);
+            CHECK(splice_writable_acceptor != nullptr) << "Splice output is not a writable acceptor";
+
+            auto* writable_acceptor = dynamic_cast<edge::IWritableAcceptor<EdgeDataTypeT>*>(&source);
+            CHECK(writable_acceptor != nullptr) << "Source is not a writable acceptor";
+
+            auto& edge_holder = dynamic_cast<edge::EdgeHolder<EdgeDataTypeT>&>(*writable_acceptor);
+            CHECK(edge_holder.check_active_connection(false)) << "No active connection to splice into";
```

Comment:
Can the dynamic cast to `edge_holder` fail and return nullptr? This could have a null reference error on the next line.